File: event-schemas/src/entities/Price.ts


class Price {
  private _amount?: number;
  private _currency?: string;
  private _currencyType?: string;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    amount?: number,
    currency?: string,
    currencyType?: string,
  }) {
    this._amount = input.amount;
    this._currency = input.currency;
    this._currencyType = input.currencyType;
  }

  get amount(): number | undefined { return this._amount; }
  set amount(amount: number | undefined) { this._amount = amount; }

  get currency(): string | undefined { return this._currency; }
  set currency(currency: string | undefined) { this._currency = currency; }

  get currencyType(): string | undefined { return this._currencyType; }
  set currencyType(currencyType: string | undefined) { this._currencyType = currencyType; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default Price;

File: event-schemas/data/csharp/Price.cs
namespace entities
{
  using System.Collections.Generic;

  public class Price
  {
    private double? amount;
    private string currency;
    private string currencyType;
    private Dictionary<string, dynamic> additionalProperties;

    public double? Amount 
    {
      get { return amount; }
      set { amount = value; }
    }

    public string Currency 
    {
      get { return currency; }
      set { currency = value; }
    }

    public string CurrencyType 
    {
      get { return currencyType; }
      set { currencyType = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}